File: level18/1874.cpp
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    int n;
    cin >> n;

    int num, idx = 0;
    stack<int> st;
    string op;
    for (int i = 0; i < n; i++){
        cin >> num;
        if (idx < num){
            while (idx != num){
                st.push(++idx);
                op += '+';
            }
            st.pop();
            op += '-';
        }
        else if (idx > num){
            if(st.empty()){
                cout << "NO\n";
                exit(0);
            }
            
            do{
                st.pop();
                op += '-';
            }while(!st.empty() && st.top() >= num);
        }
    }
    for (auto a : op){
        cout << a << "\n";
    }
}

File: baekjoon/js/level20 Stack/1874.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let [N, ...input] = fs.readFileSync(filePath).toString().trim().split("\n").map(Number);

const diff = new Array(...input);
let data;
let num = 1;

const ans = []
const stack = [];
const result = [];

while(input.length && num <=N){
  data = input.shift();

  while(num <= data){
    stack.push(num++);
    ans.push('+');
  }
  
  if(data === stack[stack.length-1]){
    result.push(stack.pop());
    ans.push('-');
  }
}

while(stack.length){
  result.push(stack.pop());
  ans.push('-');
}

diff.length === result.length && diff.every((val, idx) => val === result[idx]) ? console.log(ans.join('\n')) : console.log('NO');